SS14.Launcher/Views/HubSettingsDialog.xaml

```diff
@@ -0,0 +1,124 @@
+same game server the hub with the higher priority (higher in the list) will
+take precedence.
+        </ToolTip.Tip>
+        <ToolTip.ShowDelay>0</ToolTip.ShowDelay>
+      </Button>
+      <TextBlock DockPanel.Dock="Left" Text="" Name="Warning" Margin="4 0 0 0" Height="30" Foreground="Gray" FontSize="15" Padding="4" />
+    </DockPanel>
+    <ScrollViewer DockPanel.Dock="Top">
+      <ItemsControl Items="{Binding HubList}" Name="Hubs">
+        <ItemsControl.ItemsPanel>
+          <ItemsPanelTemplate>
+            <StackPanel Spacing="2" />
+          </ItemsPanelTemplate>
+        </ItemsControl.ItemsPanel>
+        <ItemsControl.ItemTemplate>
+          <DataTemplate>
+            <DockPanel>
+              <Button DockPanel.Dock="Right"
+                      Margin="1 0 0 0"
+                      Classes="OpenLeft"
+                      Width="30"
+                      Padding="8 0 0 0"
+                      Command="{Binding Remove}"
+                      IsEnabled="{Binding RemoveButtonEnabled}">
+                <ToolTip.Tip>Remove hub</ToolTip.Tip>
+                <Path Fill="{DynamicResource ThemeForegroundBrush}"
+                      HorizontalAlignment="Center"
+                      VerticalAlignment="Center"
+                      Data="M 0 2 L 2 0 L 6 4 L 10 0 L 12 2 L 8 6 L 12 10 L 10 12 L 6 8 L 2 12 L 0 10 L 4 6 L 0 2 Z" />
+              </Button>
+              <Button DockPanel.Dock="Right"
+                      Margin="1 0"
+                      Classes="OpenBoth"
+                      Width="30"
+                      Name="UpButton"
+                      Command="{Binding Up}">
+                <ToolTip.Tip>Increase priority</ToolTip.Tip>
+                <Path Fill="{DynamicResource ThemeForegroundBrush}"
+                      HorizontalAlignment="Center"
+                      VerticalAlignment="Center"
+                      Data="M 0 0 L 8 8 L 0 16 Z M 30 0">
+                  <Path.RenderTransform>
+                    <RotateTransform Angle="-90" />
+                  </Path.RenderTransform>
+                </Path>
+              </Button>
+              <Button DockPanel.Dock="Right"
+                      Margin="4 0 1 0"
+                      Classes="OpenRight"
+                      Width="30"
+                      Name="DownButton"
+                      Command="{Binding Down}">
+                <ToolTip.Tip>Decrease priority</ToolTip.Tip>
+                <Path Fill="{DynamicResource ThemeForegroundBrush}"
+                      HorizontalAlignment="Center"
+                      VerticalAlignment="Center"
+                      Data="M 0 0 L 8 8 L 0 16 Z M 30 0">
+                  <Path.RenderTransform>
+                    <RotateTransform Angle="90" />
+                  </Path.RenderTransform>
+                </Path>
+              </Button>
+              <TextBox Text="{Binding Address}" PropertyChanged="HubTextChanged" Watermark="https://example.com/hub/" />
```

Comment:
This probably shouldn't be a multi-line textbox?

Reply:
Are you sure it's multiline?

Reply:
Yes it is, paste some multi-line text into it.

Reply:
guh

Reply:
Since this also affects all other TextBoxes in the launcher, and I have no idea how to fix it, can I resolve this for now? Doesn't seem like blocking issue to me, a multiline string won't be a valid URI anyways unless the other lines are whitespace.

Reply:
Uhhh yeah I guess.